src/app/shared/services/authentication.service.ts

```diff
@@ -1,34 +1,54 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
 
-import { User } from '../interfaces/user.interface.';
+import { BehaviorSubject, Observable, Subject, last } from 'rxjs';
+
+import { HttpUser } from '../interfaces/http-user.interface';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
 
-  loggedUser!: User | null;
-
-  login(user: User): void {
-    this.loggedUser = {...user};
-    localStorage.setItem("token", JSON.stringify(this.loggedUser));
-    window.location.reload();
+  loggedUser!: HttpUser | null;
+  private apiUrl = 'http://localhost:3004';
+  userSubject = new Subject<HttpUser>();
+  isAuthSubject = new BehaviorSubject<boolean>(false);
+  isAuthenticated$: Observable<boolean> = this.isAuthSubject.asObservable();
+
+  constructor(private http: HttpClient) {}
+
+  login(login: string, password: string){
+    this.http
+      .post(
+        `${this.apiUrl}/auth/login`,
+        {login, password},
+      )
+      .subscribe( response => {
+        const token = JSON.parse(JSON.stringify(response)).token;
+        localStorage.setItem("token", token);
+      })
   }
 
   logout(): void {
     localStorage.removeItem("token");
     this.loggedUser = null;
-    // window.location.reload();
   }
 
   isAuthenticated(): boolean {
```

Comment:
It is better just to use subject to observe the state of the user

Reply:
Subject for authentication was done, I made a subscription for it in auth.guard